providers/celery/pyproject.toml

```
@@ -0,0 +1,91 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# NOTE! THIS FILE IS AUTOMATICALLY GENERATED AND WILL BE OVERWRITTEN!
+
+# IF YOU WANT TO MODIFY THIS FILE EXCEPT DEPENDENCIES, YOU SHOULD MODIFY THE TEMPLATE
+# `pyproject_TEMPLATE.toml.jinja2` IN the `dev/breeze/src/airflow_breeze/templates` DIRECTORY
+[build-system]
+requires = ["flit_core==3.10.1"]
+build-backend = "flit_core.buildapi"
+
+[project]
+name = "apache-airflow-providers-celery"
+version = "3.9.0"
+description = "Provider package apache-airflow-providers-celery for Apache Airflow"
+readme = "README.rst"
+authors = [
+    {name="Apache Software Foundation", email="dev@airflow.apache.org"},
+]
+maintainers = [
+    {name="Apache Software Foundation", email="dev@airflow.apache.org"},
+]
+keywords = [ "airflow-provider", "celery", "airflow", "integration" ]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Environment :: Web Environment",
+    "Intended Audience :: Developers",
+    "Intended Audience :: System Administrators",
+    "Framework :: Apache Airflow",
+    "Framework :: Apache Airflow :: Provider",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: System :: Monitoring",
+]
+requires-python = "~=3.9"
+# The dependencies should be modified in place in the generated file
+# Any change in the dependencies is preserved when the file is regenerated
+dependencies = [
+    "apache-airflow>=2.9.0",
+    # The Celery is known to introduce problems when upgraded to a MAJOR version. Airflow Core
+    # Uses Celery for CeleryExecutor, and we also know that Kubernetes Python client follows SemVer
+    # (https://docs.celeryq.dev/en/stable/contributing.html?highlight=semver#versions).
+    # Make sure that the limit here is synchronized with [celery] extra in the airflow core
+    # The 5.3.3/5.3.2 limit comes from https://github.com/celery/celery/issues/8470
+    "celery[redis]>=5.3.0,<6,!=5.3.3,!=5.3.2",
+    "flower>=1.0.0",
+    "google-re2>=1.0",
+]
+# The optional dependencies should be modified in place in the generated file
+# Any change in the dependencies is preserved when the file is regenerated
+[project.optional-dependencies]
+"cncf.kubernetes" = [
+    "apache-airflow-providers-cncf-kubernetes>=7.4.0",
+]
+
+[project.urls]
+"Documentation" = "https://airflow.apache.org/docs/apache-airflow-providers-celery/3.9.0"
+"Changelog" = "https://airflow.apache.org/docs/apache-airflow-providers-celery/3.9.0/changelog.html"
```

Should we change this from specific version to stable?
That way we don't need to change these lines with every release?
```suggestion
"Documentation" = "https://airflow.apache.org/docs/apache-airflow-providers-celery/stable"
"Changelog" = "https://airflow.apache.org/docs/apache-airflow-providers-celery/stable/changelog.html"
```

---

or maybe it's bad idea if we release version that is not the latest like we had with fab

---

No - we should not, because those links are shown by PyPI at the PyPI page for specific version of package. If we change it to stable, then links in PyPI will not be pointing to the right version docs, they will be pointing to the latest released version - no matter what version you look at.

See for example https://pypi.org/project/apache-airflow-providers-celery/3.8.5/ -> links to docs (on the left) are pointing to https://airflow.apache.org/docs/apache-airflow-providers-celery/3.8.5. not to https://airflow.apache.org/docs/apache-airflow-providers-celery/stable.

Those version number are automatically regenerated in "prepare-provider-documentation" step - we will not have to keep them updated manually.

---

Actually this is one of the main reasons why `pyproject.toml` we have is being regenerated automatically  from this template:
 https://github.com/apache/airflow/blob/93d275ef734e70cf47d217f136e599976cd76aa1/dev/breeze/src/airflow_breeze/templates/pyproject_TEMPLATE.toml.jinja2

There are few other reasons, because we want to likely keep all the pyproject.toml files for all our providers in-sync, when we will add new features/configuration options, but the "doc URL" is also an important one.

---

Ah cool. Didn't see we have the template